dbadapter/db_adapter.go

```diff
@@ -72,6 +79,33 @@ ConnectMongo:
 	logger.DbLog.Infoln("connected to MongoDB")
 }
 
+func CheckTransactionsSupport(client *DBInterface) error {
+	if client == nil || *client == nil {
+		return fmt.Errorf("mongoDB client has not been initialized")
+	}
+	ticker := time.NewTicker(60 * time.Second)
+	defer func() { ticker.Stop() }()
+	timer := time.After(180 * time.Second)
+	logger.DbLog.Infoln("checking for replica set or sharded config in MongoDB...")
+	for {
+		supportsTransactions, err := (*client).SupportsTransactions()
+		if err != nil {
+			logger.DbLog.Warnw("could not verify replica set or sharded status", "error", err)
+		}
+		if supportsTransactions {
+			break
+		}
+		select {
+		case <-ticker.C:
+			// Continue to check after each tick
+		case <-timer:
+			return fmt.Errorf("%s", "timed out while waiting for Replica Set or sharded config to be set in MongoDB")
```

Comment:
Any reason to have the error in this format (using `%s`), while in line 84 it is not used (`fmt.Errorf("mongoDB client has not been initialized")`)?
```suggestion
			return fmt.Errorf("timed out while waiting for Replica Set or sharded config to be set in MongoDB")
```

Reply:
you are right, this is fixed.